src/dash-table/dash/DataTable.js

```diff
@@ -168,34 +185,51 @@ export const propTypes = {
+         * action button on the column. If there are multiple header rows, True
+         * will display the action button on each row.
          *
-         * If True, then the column is hideable through the UI.
-         * If there are multiple column headers, only the last row of
-         * the column headers will dispay the `hide` action button.
+         * If `last`, the `hide` action button will only appear on the last header
+         * row. If `first` it will only appear on the first header row. These
+         * are respectively shortcut equivalents to `[False, ..., True]` and
+         * `[True, ..., False]`.
          *
-         * Hidden columns are listed by id in the `hidden_columns` prop.
+         * If there are merged, multi-header columns then you can choose
+         * which column header row to display the `hide` action button in by
+         * supplying an array of booleans.
+         * For example, `[true, false]` will display the `hide` action button
+         * on the first row, but not the second row.
          *
-         * If `hidden_columns` is not empty, a toggle button is displayed
-         * above the table that allows toggling columns visibility.
+         * If the `hide` action button appears on a merged column, then clicking
+         * on that button will hide *all* of the merged columns associated with it.
          */
-        hideable: PropTypes.bool,
+        hideable: PropTypes.oneOfType([
+            PropTypes.oneOf(['first', 'last']),
+            PropTypes.bool,
+            PropTypes.arrayOf(PropTypes.bool)
+        ]),
 
         /**
-         * If True, then the name of this column is editable.
-         * If there are multiple column headers (if `name` is a list of strings),
-         * then `renamable` can refer to _which_ column header should be
-         * editable by defining an array of booleans.
-         * For example, `[true, false]` will make the first row's name editable,
-         * but not the second row.
-         * Also, updating the name in a merged column header cell will
-         * update the name of each column.
+         * If True, the user can rename the column by clicking on the `rename`
+         * action button on the column. If there are multiple header rows, True
+         * will display the action button on each row.
+         *
+         * If `last`, the `rename` action button will only appear on the last header
+         * row. If `first` it will only appear on the first header row. These
+         * are respectively shortcut equivalents to `[False, ..., True]` and
+         * `[True, ..., False]`.
+         *
+         * If there are merged, multi-header columns then you can choose
+         * which column header row to display the `rename` action button in by
+         * supplying an array of booleans.
+         * For example, `[true, false]` will display the `rename` action button
+         * on the first row, but not the second row.
```

Comment:
Are we using Python (`True`) or JS (`true`) capitalization here? This inconsistency exists all sorts of places across our component repos, but IIRC we said we'd try to standardize on JS in the JS code, paving the way for language-dependent translation at some future time in the back-end codegen.

Also `[false, ..., true]` is ambiguous. Perhaps `[false, ..., false, true]`? That said I might just remove the shortcut sentence, as it's not a fixed shortcut (depends on # of headers) and I think it's already pretty clear without that.

Reply:
Will improve consistency for the modified props and go the JS route. Good point for the array examples.